variables.tf

```diff
@@ -1 +1,243 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# REQUIRED PARAMETERS
+# You must provide a value for each of these parameters.
+# ---------------------------------------------------------------------------------------------------------------------
 
+variable "name" {
+  type        = string
+  description = "Name of the vault stack, will be use to prefix resources"
+}
+
+variable "environment" {
+  type        = string
+  description = "Type of environment -- must be one of: dev, test, prod"
+}
+
+variable "key_pair_name" {
+  type        = string
+  description = "Keypair to associate to launched instances"
+}
+
+variable "ami_owners" {
+  type        = list(string)
+  description = "Account id/alias of the AMI owners"
+}
+
+variable "ec2_extra_security_group_ids" {
+  type        = list(string)
+  description = "List of additional security groups to add to EC2 instances"
+  default     = []
+}
+
+variable "ec2_subnet_ids" {
+  type        = list(string)
+  description = "List of subnets where EC2 instances will be launched"
+}
+
+variable "lb_subnet_ids" {
+  type        = list(string)
+  description = "List of subnets to associate to the Load Balancer"
+}
+
+variable "vault_version" {
+  type        = string
+  description = "Version of Vault to be installed on servers"
+}
+
+variable "vault_url" {
+  type        = string
+  description = "The DNS address that vault will be accessible at. Stack name will be used as the url when value is set to empty. Example: vault.domain.net"
+  default     = null
+}
+
+variable "domain_name" {
+  type        = string
+  description = "The domain name where vault url will be registered to. Example: domain.net"
+}
+
+variable "route53_zone_id" {
+  type        = string
+  description = "Hosted zone ID Route 53 hosted zone"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# OPTIONAL PARAMETERS
+# These parameters have reasonable defaults.
+# ---------------------------------------------------------------------------------------------------------------------
+variable "kms_key_id" {
+  type        = string
+  description = "Id of an AWS KMS key use for auto unseal operation when vault is intialize"
+  default     = null
+}
+
+variable "dynamodb_table" {
+  type        = string
+  description = "Name of the Dynamodb to be used as storage backend for Vault"
+  default     = null
+}
+
+variable "ami_name_filters" {
+  type        = list(string)
+  description = "Will be use to filter out AMI"
+  default     = ["spel-minimal-centos-7-hvm-*.x86_64-gp2"]
+}
+
+variable "ami_name_regex" {
+  type        = string
+  description = "Regex to help fine-grain filtering AMI"
+  default     = "spel-minimal-centos-7-hvm-\\d{4}\\.\\d{2}\\.\\d{1}\\.x86_64-gp2"
+}
+
+variable "vault_configs_path" {
```

Comment:
gonna need some real documentation on how this works and what configs/options are supported